File: models/userdata.js
const mongoose = require('mongoose');

const userSchema = new mongoose.Schema({
  restaurantName: { type: String, required: true },
  restaurantAddress: { type: String, required: true },
  phoneNumber: { type: String, required: true, unique: true },
  alternateNumber: { type: String, unique: true },
  email: { type: String, required: true, unique: true },
  password: { type: String, required: true },
  ownerName: { type: String, required: true },
  ownerPhoneNumber: { type: String, required: true, unique: true },
  ownerEmail: { type: String, unique: true },
  receiveUpdatesOnWhatsApp: { type: Boolean }
});

module.exports = mongoose.model('User', userSchema);

File: app-code/Preperly/app/src/main/java/com/example/preperly/datamodels/UserData.kt
package com.example.preperly.datamodels

data class User(
    val restaurantName: String,
    val restaurantAddress: String,
    val phoneNumber: String,
    val alternateNumber: String,
    val email: String,
    val password: String,
    val ownerName: String,
    val ownerPhoneNumber: String,
    val ownerEmail: String,
    val receiveUpdatesOnWhatsApp: Boolean
)
